hive-metastore/src/main/java/org/apache/iceberg/hive/CachedClientPool.java

```diff
@@ -87,4 +102,89 @@ public <R> R run(Action<R, IMetaStoreClient, TException> action, boolean retry)
       throws TException, InterruptedException {
     return clientPool().run(action, retry);
   }
+
+  @VisibleForTesting
+  static Key extractKey(String cacheKeys, Configuration conf) {
+    // generate key elements in a certain order, so that the Key instances are comparable
+    List<Object> elements = Lists.newArrayList();
+    elements.add(conf.get(HiveConf.ConfVars.METASTOREURIS.varname, ""));
+    if (cacheKeys == null || cacheKeys.isEmpty()) {
```

Comment:
Selfishly, I would add 'default.catalog' to here as well, as I'm not exactly sure any use case where Iceberg re-uses HMS Client with different catalogs.  (As we never allow user to pass in catalog explicitly to HMSClient).  But I'm ok to do it in another pr which we can contribute, for readability.

Reply:
OK let's leave it to another PR.